create: updated to oauth

bnert opened the pull request.

Commit:
create: updated to oauth

yejingyu commented on src/jama/create/create.py:
```diff
@@ -46,10 +49,17 @@ def from_dialog(url_base, json_to_parse):
 
-        jama_resp = requests.post(jama_url, json=to_post_obj, auth=(os.environ['JAMA_USER'], os.environ['JAMA_PASS']))
-            
+        if "DB_HOST" not in os.environ:
```
You can simply use the `api_caller.post()` in `jama/api_caller` to do the same thing.
I think using the same caller function can make our code more maintainable.

yejingyu commented on src/jama/create/create.py:
```diff
@@ -143,8 +153,15 @@ def from_text(base_url, content_dict):
     url = base_url + '/rest/latest/items/'
-    jama_resp = requests.post(
-        url, json=to_post_obj, auth=(os.environ['JAMA_USER'], os.environ['JAMA_PASS']))
+    if "DB_HOST" not in os.environ:
```
Similar to line 53.
You can simply use the `api_caller.post()` in `Jama/api_caller` to do the same thing.
I think using the same caller function can make our code more maintainable.

bnert replied:
@yejingyu done

yejingyu reviewed (dismissed):
Is there any reason of why we don't use the api_caller.post()?

Commit:
create: updated dialog boxes

The dialog box now automatically fills in the item type.

All you need to do is specify a name and a description.
